tools/pythonpkg/tests/arrow/test_parallel.py

```diff
@@ -0,0 +1,24 @@
+import duckdb
+import os
+import sys
+try:
+    import pyarrow
+    import pyarrow.parquet
+    import numpy as np
+    can_run = True
+except:
+    can_run = False
+
+class TestArrowParallel(object):
+    def test_parallel_run(self,duckdb_cursor):
+        if not can_run:
+            return
+        duckdb_conn = duckdb.connect()
+        duckdb_conn.execute("PRAGMA threads=4")
+        duckdb_conn.execute("PRAGMA force_parallelism")
+        data = (pyarrow.array(np.random.randint(800, size=1000000), type=pyarrow.int32()))
+        tbl = pyarrow.Table.from_batches(pyarrow.Table.from_arrays([data],['a']).to_batches(10000))
```

Comment:
Perhaps add some more tests:
* Test with different numbers of batches
* More types
* More complex queries (e.g. a join between two arrow scans)
* Parallel scans on small tables with fewer batches than threads
* Test with an error condition coming from the stream

Reply:
I've added #1720 for this. This PR would grow in complexity with things non-related to parallelism (e.g., joins are not working r-now).